Neos.ContentRepository.Core/Classes/Projection/Workspace/Workspace.php

```diff
@@ -24,31 +24,72 @@
  *
  * @api
  */
-class Workspace
+final readonly class Workspace
 {
     /**
      * This prefix determines if a given workspace (name) is a user workspace.
      */
     public const PERSONAL_WORKSPACE_PREFIX = 'user-';
 
+    /**
```

Comment:
Note: The following properties have been turned into actual fields to improve IDE support:
Constructor promoted arguments of an `@internal` constructor are all "internal", but in this case they are _not_. Besides the `@deprecated` tags are ignored otherwise